Formula/p/poutine.rb

```diff
@@ -0,0 +1,34 @@
+class Poutine < Formula
+  desc "Security scanner that detects vulnerabilities in build pipelines"
+  homepage "https://boostsecurityio.github.io/poutine/"
+  url "https://github.com/boostsecurityio/poutine/archive/refs/tags/v0.10.1.tar.gz"
+  sha256 "ad9fb2dc3ca35e0606aea6059d38a36552ab68643c79955deb1985f302d5aa91"
+  license "Apache-2.0"
+  head "https://github.com/boostsecurityio/poutine.git", branch: "main"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{tap.user}
+      -X main.date=#{time.iso8601}
+    ]
+
+    system "go", "build", *std_go_args(ldflags:)
+
+    generate_completions_from_executable(bin/"poutine", "completion")
+  end
+
+  test do
+    output = shell_output("#{bin}/poutine help")
+    assert_match "analyze_org   Analyzes an organization's repositories for supply chain vulnerabilities", output
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.